solr/core/src/java/org/apache/solr/core/NodeConfig.java

```diff
@@ -929,8 +941,8 @@ public NodeConfig build() {
           metricsConfig,
           cachesConfig,
           tracerConfig,
-          containerPluginsSource,
-          containerPlugins,
+          clusterPluginsSource,
+          clusterSingletonPlugins,
```

Comment:
In NodeConfig, do we really need each supported type separated out?  It's only "clusterSingleton" now but there are others that will be added.  I could see simply an array of them or maybe a map keyed by name/type and value is the array.

Come to think of it, I wonder if we could migrate all PluginInfo used in NodeConfig to use this Map!  This would certainly majorly cut down on NodeConfig's Builder!  If we do such a thing, it'd probably be best in a separate PR (before or after).

Reply:
CC @janhoy

Reply:
You know I like simplifications. So as long as they are handled in the same generic way, it would be a welcome change. The devil could be in the details when one dives into the task though, so let's postpone it to a separate investigation.